src/bitdrift_public/fbs/issue-reporting/v1/report.fbs

```diff
@@ -0,0 +1,298 @@
+    // - locked resource causing thread blocked state
+    // - awaited thread ID
+    state:[string];
+}
+
+table Thread {
+    name:string;
+
+    // true if this thread is reporting the problem
+    active:bool = false;
+
+    // numeric identifier for the thread, such as Java's Thread.getId()
+    index:uint32;
+
+    // platform-specific name of the thread's current operation mode
+    state:string;
+
+    // platform-specific value indicating the relative priority of a thread
+    // Apple platforms: value between 0.0 and 1.0
+    // JVM: integer value between 1 and 10
+    // .NET: integer value between 0 and 4
+    priority:float;
+
+    // (Apple platforms) the level of resources available to the thread
+    quality_of_service:int8 = -1;
+
+    // execution path, ordered from most recently executed to least recently.
+    // (i.e. a main() function would be last not first)
+    stack_trace:[Frame];
+}
+
+table ThreadDetails {
+    // total number of threads, which may exceed the number able to be
+    // captured in the threads collection
+    count:uint16;
+
+    threads:[Thread];
+}
+
+table Error {
+    // Descriptive category for the event, such as the fully-qualified name of
+    // an exception, Mach or POSIX signal, or termination category (e.g.
+    // "Application Not Responding", "Memory Pressure Termination", "StrictMode
+    // Violation")
+    name:string;
+
+    // Contextual information about the event, briefly explaining why the event
+    // happened. Frequently the message content of the Exception or Error
+    // instance, if any.
+    reason:string;
+
+    // execution path, ordered from most recently executed to least recently.
+    // (i.e. a main() function would be last not first)
+    stack_trace:[Frame];
+
+    // Given there is a subsequent entry in the collection containing this Error,
+    // it has the following relation to this Error.
+    relation_to_next:ErrorRelation = CausedBy;
+}
+
+table BinaryImage {
+    // identifier for the compiled binary - corresponds to Frame.image_id
+    id:string;
+
+    // path on disk at build time
+    path:string;
+
+    // offset between the start of program memory and this image
+    load_address:uint64;
+}
+
+table SDKInfo {
+    // identifier for a build configuration generating a report, i.e. capture-sdk
+    // on Android being io.bitdrift.capture-android
+    id:string;
+
+    // release version
+    version:string;
+}
+
+table Report {
+    sdk:SDKInfo;
+
+    app_metrics:AppMetrics;
+
+    device_metrics:DeviceMetrics;
+
+    // The captured Error (or Exception), where the first entry is the error
+    // which was captured or reported, and subsequent entries are related to
+    // the first by a specified relation, such as being the cause.
+    errors:[Error];
+
+    // thread state at the moment the event occurred or was reported. No
+    // guaranteed ordering between the threads in the collection.
+    thread_details:ThreadDetails;
```

Comment:
do we have a way to associate each error/stack trace, to a thread? is it just by index?